client/service-container-chain/src/data_preservers.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+use {
+    crate::spawner::{ContainerChainSpawner, TSelectSyncMode},
+    dc_orchestrator_chain_interface::{
+        DataPreserverAssignment, OrchestratorChainInterface, OrchestratorChainResult,
+    },
+    futures::stream::StreamExt,
+    std::{future::Future},
+    tc_consensus::ParaId,
+};
+
+pub type ProfileId = <dancebox_runtime::Runtime as pallet_data_preservers::Config>::ProfileId;
```

Comment:
I would try to make this generic, i.e. for any runtime that implements `pallet_data_preservers::Config`, I am thinking of starlight here

Reply:
Since it interacts with the orchestrator interface it should probably use this type:
https://github.com/moondance-labs/dancekit/pull/32/files#diff-cc0640977493f1e56ff4eec9045ead68475bd6d2d41284ffa919745c7e726bb5R101

The linked type cannot be generic since we can't know in advance to what runtime we're connecting to I think.

Reply:
This is not needed, somehow I managed to support starlight without adding starlight as a dependency.

Reply:
alright, then let's disregard